src/app/components/board/board.component.ts

```diff
@@ -0,0 +1,94 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { Board } from 'src/app/constants/boardClass';
+import { BoardService } from 'src/app/services/board.service';
+import { ColumnService } from 'src/app/services/column.service';
+import { Column } from 'src/assets/interfaces/Column';
+
+@Component({
+  selector: 'app-board',
+  templateUrl: './board.component.html',
+  styleUrls: ['./board.component.scss'],
+  providers: [BoardService],
+})
+export class BoardComponent implements OnInit {
+  board!: Board;
+  showModal: boolean = false;
+  showCreateColumnModal: boolean = false;
+  showCreateTaskModal: boolean = false;
+  newColumnName: string = '';
+
+  constructor(
+    private route: ActivatedRoute,
+    private boardService: BoardService,
+    private columnService: ColumnService,
+    private router: Router
+  ) {}
+
+  ngOnInit() {
+    const boardId = this.route.snapshot.paramMap.get('_id');
+    localStorage.setItem('boardId', boardId ?? '');
+    if (boardId) {
+      this.boardService.getBoardById(boardId).subscribe({
```

Comment:
Every time visiting this page will be created a new subscription. This leads to performance issue. So you should pay attention to closing them. It might be unsubscribing ( in the onDestroy method) or approaches from the article https://blog.bitsrc.io/6-ways-to-unsubscribe-from-observables-in-angular-ab912819a78f. In the projects' life, we use take* more often than others.